common/test/acceptance/pages/common/utils.py

```diff
@@ -1,35 +1,57 @@
 """
 Utility methods common to Studio and the LMS.
 """
-from bok_choy.promise import EmptyPromise
+from bok_choy.promise import BrokenPromise
 from common.test.acceptance.tests.helpers import disable_animations
 from selenium.webdriver.common.action_chains import ActionChains
 
 
-def wait_for_notification(page):
+def sync_on_notification(page, style='default', wait_for_hide=False):
     """
-    Waits for the "mini-notification" to appear and disappear on the given page (subclass of PageObject).
+    Sync on notifications but do not raise errors.
+
+    A BrokenPromise in the wait_for probably means that we missed it.
+    We should just swallow this error and not raise it for reasons including:
+    * We are not specifically testing this functionality
+    * This functionality is covered by unit tests
+    * This verification method is prone to flakiness
+      and browser version dependencies
+
+    See classes in edx-platform:
+     lms/static/sass/elements/_system-feedback.scss
     """
-    def _is_saving():
-        """Whether or not the notification is currently showing."""
-        return page.q(css='.wrapper-notification-mini.is-shown').present
-
-    def _is_saving_done():
-        """Whether or not the notification is finished showing."""
-        return page.q(css='.wrapper-notification-mini.is-hiding').present
-
-    EmptyPromise(
-        _is_saving,
-        'Notification should have been shown.',
-        try_interval=0.1,
-        timeout=60,
-    ).fulfill()
-    EmptyPromise(
-        _is_saving_done,
-        'Notification should have been hidden.',
-        try_interval=0.1,
-        timeout=60,
-    ).fulfill()
+    hiding_class = 'is-hiding'
+    shown_class = 'is-shown'
+
+    def notification_has_class(style, el_class):
+        """
+        Return a boolean representing whether
+        the notification has the class applied.
+        """
+        if style == 'mini':
+            css_string = '.wrapper-notification-mini.{}'
+        else:
+            css_string = '.wrapper-notification-confirmation.{}'
+        return page.q(css=css_string.format(el_class)).present
+
+    # Wait for the notification to show.
+    # If you miss it though, don't raise an error.
```

Comment:
It is at an acceptable speed (manually verified), but when browser isn't displayed (while running tests), it is sporadic